bokehjs/src/coffee/models/renderers/glyph_renderer.coffee

```diff
@@ -230,7 +230,10 @@ export class GlyphRendererView extends RendererView
     @glyph.draw_legend_for_index(ctx, x0, x1, y0, y1, index)
 
   hit_test: (geometry) ->
-    @glyph.hit_test(geometry)
+    if @model.visible
+      return @glyph.hit_test(geometry)
+    else
+      return null
```

Comment:
An option (to make testing simpler) would be to have this just call a new `GlyphRenderer.hit_test`) that does what the code above does. That could be tested in the manner described.